### PR TITLE
virtualization_resource: Fix `NoMethodError` on `nil:NilClass`

### LINKED ISSUE #2532: `inspec check` on a non-virtualized host stacktraces if referencing virtualization.system
### Description

Running `inspec check` on a non-virtualized host results in a stack trace if the test has a conditional block such as:

```
if virtualization.system != 'docker'
 ...
end
```

### InSpec and Platform Version

```
$ bundle exec inspec --version
1.51.0
$ ruby --version
ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin16]
```

### Replication Case

Run `inspec init profile testcase` and copy this in place of `example.rb`:
```
if virtualization.system != 'docker'

  control 'testcase-1' do
    impact 0.7
    title 'Recreate stack trace'
    desc 'Recreate stack trace'

    describe file('/etc/passwd') do
      it { should exist }
    end
  end
end
```

Then run `inspec check testcase` and you'll get a stacktrace 

### Possible Solutions

Add some error checking to virtualization.rb so it doesn't throw an undefined method error when not on a virtualized platform

### Stacktrace

```
/usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/inspec-1.51.0/lib/resources/virtualization.rb:39:in `block (2 levels) in <class:Virtualization>': undefined method `[]' for nil:NilClass (NoMethodError)
	from testcase/controls/example.rb:1:in `load_with_context'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/inspec-1.51.0/lib/inspec/profile_context.rb:157:in `instance_eval'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/inspec-1.51.0/lib/inspec/profile_context.rb:157:in `load_with_context'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/inspec-1.51.0/lib/inspec/profile_context.rb:141:in `load_control_file'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/inspec-1.51.0/lib/inspec/profile.rb:167:in `block in collect_tests'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/inspec-1.51.0/lib/inspec/profile.rb:164:in `each'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/inspec-1.51.0/lib/inspec/profile.rb:164:in `collect_tests'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/inspec-1.51.0/lib/inspec/profile.rb:470:in `load_checks_params'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/inspec-1.51.0/lib/inspec/profile.rb:463:in `load_params'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/inspec-1.51.0/lib/inspec/profile.rb:157:in `params'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/inspec-1.51.0/lib/inspec/profile.rb:323:in `controls_count'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/inspec-1.51.0/lib/inspec/profile.rb:294:in `check'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/inspec-1.51.0/lib/inspec/cli.rb:70:in `check'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor/command.rb:27:in `run'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor/invocation.rb:126:in `invoke_command'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor.rb:387:in `dispatch'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/thor-0.20.0/lib/thor/base.rb:466:in `start'
	from /usr/local/var/rbenv/versions/2.4.1/lib/ruby/gems/2.4.0/gems/inspec-1.51.0/bin/inspec:12:in `<top (required)>'
	from /usr/local/var/rbenv/versions/2.4.1/bin/inspec:22:in `load'
	from /usr/local/var/rbenv/versions/2.4.1/bin/inspec:22:in `<main>'
```

### PR DESCRIPTION
**Pull request by jerryaldrichiii**
This moves the creation of `@virtualization_data` to `initialize`.

Methods for `role` and `system` properties are dynamically generated and return values from the `@virtualization_data` Mash. Therefor, we must ensure `@virtualization_data` exists before calling these methods.

This removes an `author` line since the Git history serves as the authoritative source for who was involved in development.

This adds rudimentary tests. I have created issue https://github.com/chef/inspec/issues/2602 to have more tests added.

This closes #2532.

### CODE REVIEW

**Review by miah — APPROVED** [review]

File: lib/resources/virtualization.rb [hunk 1]
```diff
@@ -25,11 +24,14 @@ class Virtualization < Inspec.resource(1)
     "
 
     def initialize
+      @virtualization_data = Hashie::Mash.new
+
       unless inspec.os.linux?
-        skip_resource 'The `virtualization` resource is not supported on your OS yet.'
-      else
-        collect_data_linux
+        raise Inspec::Exceptions::ResourceSkipped,
```
Comment (jquick) [inline lib/resources/virtualization.rb]: instead of this can we just put `supports platform: 'linux'` at the top?

**Review by jquick — COMMENTED** [review]
Thanks @jerryaldrichiii. Just one small thing to move the supports.

File: lib/resources/virtualization.rb (hunk 1 above)
Comment (jerryaldrichiii) [reply lib/resources/virtualization.rb]: Looks like `supports: unix` is on master. Will remove these bits and rebase.

**Review by jquick — APPROVED** [review]
Thanks @jerryaldrichiii !

**Pull request merged**